zathura/zathura.h

```diff
@@ -223,6 +227,15 @@ struct zathura_s {
    * Context for MIME type detection
    */
   zathura_content_type_context_t* content_type_context;
+
+  /**
+   * SyncTeX context. The scanner object is cached for better performance.
+   */
+  struct {
+    synctex_scanner_p scanner;
+    time_t last_modification_time; // the last modification time of the synctex file
```

Comment:
This doesn't need to be tracked. The idea would be that if the file is reloaded due to changes, the scanner is freed. The next time the scanner is used, it can be reopened.

Reply:
Makes sense.

Do you think it's worth considering the possibility that (somehow) only the synctex file is updated and not the PDF (e.g. the user surgically modify the synctex file somehow)?

What I think: in practice it will never happen, and the most common cases (e.g. the user delete the synctex file, a new synctex file appear because user move it from another directory) are gracefully handled.

Reply:
Alright, done.

Remark: `sc_reload` is called on PDF file change and `zathura_free` is called on program exit, but both of them calls `document_close` anyway.

Side note 2, technically the non-null check is redundant because `synctex_scanner_free` does the check anyway.